### DISCUSSION TITLE
Provisioning: Lost ability to sync from git

### DISCUSSION THREAD

**Discussion by bitcollector1** [netbox-community/netbox#13616] [Help Wanted!] [answered]
### NetBox version

v3.6-beta1

### Python version

3.8

### Steps to Reproduce

Lost the ability to sync from GIT.

<img width="668" alt="Screenshot 2023-08-30 at 11 52 16" src="https://github.com/netbox-community/netbox/assets/50723251/f00b7576-8cd3-4b97-bbcd-3a04c7281216">

Here is the config template that synced successfully last week (I believe it was before the upgrade) 
<img width="551" alt="Screenshot 2023-08-30 at 11 47 48" src="https://github.com/netbox-community/netbox/assets/50723251/129dee8e-bd55-4413-b3d3-74bdc26cb820">

### Expected Behavior

Template syncs successfully as before. We have another instance not on the newest beta that syncs just fine. 

### Observed Behavior

Error when clicking on the template:
<img width="777" alt="Screenshot 2023-08-30 at 11 47 41" src="https://github.com/netbox-community/netbox/assets/50723251/4450d377-4760-490b-a7e9-fe27124e4d3a">

**Comment by jeremystretch** [comment]
Moving this to a discussion as you have not provided any reproduction instructions.

**Comment by bitcollector1** [comment]
Not sure what else to add honestly, I just noticed that the sync failed today, it was previously working before I upgraded the beta version.

**Comment by bitcollector1** [comment]
ZTP just failed on the new box I just updated to 3.6......it was working previously. 

<img width="778" alt="Screenshot 2023-08-30 at 16 33 20" src="https://github.com/netbox-community/netbox/assets/50723251/209b2efc-f9a2-4297-a954-f534ced1884c">

**Comment by bitcollector1** [comment]
Really feeling like this is a bug....as you can see the successful provision happened earlier today, pre 3.6 update. 

<img width="697" alt="Screenshot 2023-08-30 at 18 25 46" src="https://github.com/netbox-community/netbox/assets/50723251/8949df09-4fed-44b0-8d76-0465cc72bf1e">

**Comment by chuegel** [comment]
I'm experiencing this on NetBox 3.5.7 on k8s. See below: 

https://github.com/netbox-community/netbox/issues/13573

**Comment by bitcollector1** [comment]
Thanks for the heads up!  

In my case the sync was working fine on 3.5.7 but the issue popped up the second I upgraded to 3.6. (on 2 different boxes with different data). 

Hopefully it's an easy fix because I use the git sync pretty heavily.

**Comment by zndrr** [comment]
Ah I also have experienced it on v3.6-beta1, beta2 and now v3.6.0. Worked at least between v3.5.6-3.5.8, possibly before but can't remember.
Hadn't had time to collate info for a bug report sorry.

All I've done is re-add repos, but one of the ones I had a couple of the Netbox-community ones, such as: 
https://github.com/netbox-community/customizations, branch master, no auth

EDIT: yeah, definitely okay on 3.5.8 . I cloned that for 3.6b1, then upgraded through 3.6b2 and finally 3.6.0. Not fresh, if that matters.

**Comment by kkthxbye-code** [comment, marked as answer]
For some reason this wasn't pointed out in the release notes, but dulwich (git) and boto3 (aws) were removed as default packages. 

See https://docs.netbox.dev/en/stable/features/synchronized-data/

>
> Data backends which connect to external sources typically require the installation of one or more supporting Python libraries. The Git backend requires the [dulwich](https://www.dulwich.io/) package, and the S3 backend requires the [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html) package. These must be installed within NetBox's environment to enable these backends.

And https://docs.netbox.dev/en/stable/installation/3-netbox/#remote-data-sources

> NetBox supports integration with several remote data sources via configurable backends. Each of these requires the installation of one or more additional libraries.

> Amazon S3: [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html)
> Git: [dulwich](https://www.dulwich.io/)
> For example, to enable the Amazon S3 backend, add boto3 to your local requirements file:

> `sudo sh -c "echo 'boto3' >> /opt/netbox/local_requirements.txt"`


This might be the issue you are facing.

**Reply by zndrr** [reply]
Yeah that did the trick thanks. Makes sense that that addendum might have been easy to miss.

**Reply by bitcollector1** [reply]
Thank you VERY much!

**Comment by chris240189** [comment]
> ZTP just failed on the new box I just updated to 3.6......it was working previously.
> <img alt="Screenshot 2023-08-30 at 16 33 20" width="778" src="https://user-images.githubusercontent.com/50723251/264495911-209b2efc-f9a2-4297-a954-f534ced1884c.png">

I get exactly that error and I am not even using git, but a local datasource. Any ideas?
My files seem to get synced, but I cannot look at them through GUI and another plugin I was trying out (config diffs) cannot work with the datasource.

[I created an issue for tracking it](https://github.com/netbox-community/netbox/issues/13656)

**Comment by bitcollector1** [comment]
My sync jobs are still failing, even after installing the git package :(

**Reply by abhi1693** [reply]
It could be due to https://github.com/netbox-community/netbox/issues/13656

**Reply by bitcollector1** [reply]
Thank you for the assist, it ended up being a couple of different issues but happy to report it's now all working!

**Comment by bitcollector1** [comment]
I can now read the files when I click on them (after making the update in #13656 

However, the SYNC is still failing :(

**Comment by bitcollector1** [comment]
I've given up for the day, I'll try to pull again later in the week :(

**Comment by bitcollector1** [comment]
seeing the following errors on install
```
Running migrations:
  No migrations to apply.
  Your models in app(s): 'circuits', 'core', 'dcim', 'extras', 'ipam', 'tenancy', 'users', 'virtualization', 'wireless' have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
```

**Comment by bitcollector1** [comment]
Now it looks like the sync happens but the status still says failed! 

<img width="689" alt="nb-git" src="https://github.com/netbox-community/netbox/assets/50723251/38f3474a-e387-4241-8b55-8851d99358df">

**Comment by bitcollector1** [comment]
okay, I was able to confirm that the new templates have, in fact, been pulled. It appears the status message is incorrect now for whatever reason.

**Comment by bitcollector1** [comment]
okay, I FINALLY GOT IT ALL WORKING :) 

It was a combo of the (2) issues and the order of operations of my hacks resulted in a mixed state. 

After manually installing dulwich and also making the edit to the data file I'm finally back to working :) 

<img width="668" alt="Screenshot 2023-09-01 at 08 54 24" src="https://github.com/netbox-community/netbox/assets/50723251/03e121d9-2ec9-4763-96de-bd9bcb909079">
